hudi-common/src/main/java/org/apache/hudi/common/model/DefaultHoodieRecordPayload.java

```diff
@@ -45,12 +45,12 @@ public class DefaultHoodieRecordPayload extends OverwriteWithLatestAvroPayload {
   public static final String DELETE_MARKER = "hoodie.payload.delete.marker";
   private Option<Object> eventTime = Option.empty();
 
-  public DefaultHoodieRecordPayload(GenericRecord record, Comparable orderingVal) {
-    super(record, orderingVal);
+  public DefaultHoodieRecordPayload(GenericRecord record, Comparable orderingVal, Properties props) {
+    super(record, orderingVal, props);
   }
```

Comment:
The source of the props seems a chaos, I already saw several ways how it was produced:

1. `config.getPayloadConfig().getProps()` in `HoodieMergeHandle`;
2. `payloadProps.setProperty(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY, preCombineField);` in `HoodieFileSliceReader`;
3. `config.getProps()` in `HoodieIndexUtils`.

Reply:
Yeah, this should be cleaned up separately.  Regardless, the payload-related configs should always be included so there's no correctness issue.  And we're not storing any props with the record so record processing is consistent.